### PR TITLE
[MM-20487] Migrate 'components/post_view/post_time' module and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by jakubnovak998**
#### Summary
- Rename the files to their associated TypeScript extensions (ie. js to ts, jsx to tsx)
- Update any components using those to the correct imports
- Fix any errors generated by the TypeScript compiler. You can run make check-types to display any errors.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/15454

### CODE REVIEW

**Comment by jasonblais** [comment]
Thanks @jakubnovak998 for the contribution! :)

File: components/post_view/post_time/index.ts [hunk 1]
```diff
@@ -4,12 +4,14 @@
...
+import PostTime from './post_time';
+
+function mapStateToProps(state: GlobalState) {
     return {
         teamUrl: getCurrentRelativeTeamUrl(state),
     };
 }
 
-export default connect(mapStateToProps)(PostTime);
+export default connect(mapStateToProps)(PostTime as any);
```
Comment (nevyangelova) [inline components/post_view/post_time/index.ts:17]: Why do we need this casting?

Comment (jakubnovak998) [reply components/post_view/post_time/index.ts:17]: There was a ts error that I didn't know how to fix: Error:(17, 41) TS2345: Argument of type 'typeof PostTime' is not assignable to parameter of type 'ComponentType<never>'.
  Type 'typeof PostTime' is not assignable to type 'ComponentClass<never, any>'.
    Types of property 'defaultProps' are incompatible.
      Type '{ eventTime: number; location: string; }' is not assignable to type 'undefined'.

**Comment by jasonblais** [comment]
@nevyangelova @sudheerDev kind reminder to help with a review of this PR :)

File: components/post_view/post_time/index.ts (hunk 1 above)
Comment (nevyangelova) [reply components/post_view/post_time/index.ts:17]: @jakubnovak998 Your component expects no props, so your connect has an error as it infers that `mapStateToProps` should return an empty object. Try typing the defaultProps object and creating an interface 
```
type defaultProps = {
    eventTime: number,
    location: \\ whatever the location type is
}

export default class PostTime extends React.PureComponent<Props & defaultProps> {
   \\ ...
}
```

**Review by nevyangelova — CHANGES_REQUESTED** [review]
@devinbinnie can you advice here?

File: components/post_view/post_time/index.ts (hunk 1 above)
Comment (devinbinnie) [reply components/post_view/post_time/index.ts:17]: @jakubnovak998 Can you post the error you're getting without casting as `any`?

**Review by devinbinnie — APPROVED** [review]
Thanks @jakubnovak998, this is looking good now!

**Review by nevyangelova — APPROVED** [review]

**Review by lindy65 — APPROVED** [review]
Thanks @jakubnovak998 - this will be QA'd after merge.

@devinbinnie - please help add test steps to the [Jira ticket](https://mattermost.atlassian.net/browse/MM-20487), Thanks!

**Comment by devinbinnie** [comment]
@jakubnovak998 Can you resolve conflicts for us? We can merge after that.

**Pull request merged**